Update XIP-0 (#29)
xmtp/XIPs: XIPs/xip-0-purpose-process.md

## XIP Workflow

## Getting started

It's important to gather feedback about your proposal or new idea within the community before advancing it further in the process. This helps to vet your idea before going through the formality of the Improvement Proposal process.

> Engaging with the XMTP community first will help determine the originality of an idea, and will help to avoid wasting time on something that may be rejected based on prior research.

Reviewer: engaging with the community first is important to assess demand for an idea and alignment over the solution - it's not just about originality

Author: @saulmc how's this?

```suggestion
Engaging with the XMTP community first will help gauge interest in the idea, determine its originality, and will help to avoid spending time on something that may be rejected based on prior research.
```

> **It's recommended to create a new topic in the [XIP Ideas forum](https://github.com/orgs/xmtp/discussions/categories/xip-ideas) to do this.**

Reviewer 2: Maybe a note about prototyping?

> You may also want to consider prototyping the idea to derisk it and provide a reference implementation to anchor the discussions around.

### Shepherding an XIP

> Parties involved in the process are you, the champion or *XIP author*, the [*XIP editors*](#xip-editors), and the *XMTP Core Developers* (currently the XMTP Labs team).

Reviewer 3: What is the relationship between XIPs and proposed content types? Content types do not require network wide changes and do not require input from XMTP Core Developers/XIP Editors to be deployed; nor do they fit into the XIP types listed [here](https://github.com/xmtp/XIPs/pull/29/files#diff-d7e65a38761669618c147566399dc70051a93e1f785c1f6853d60e39d1b8b828R26-R32). 

Are New content type explicitly out of scope for an XIP?

Reviewer: @jazzz according to XIP-0 they fall under the XRC category of XIP

> XRC (XMTP Request for Comment): includes application-level standards and conventions, including message payload formats and applications

Reviewer 4: Maybe answering this is not in the scope of this PR, @galligan, but wanted to surface this question as I see @jazzz is curious about a similar topic.

I've seen XIPs positioned as: ”If it doesn’t exist, implement it. If you believe the community can adopt your implementation as a standard, consider proposing an XIP. Would like to see people, experimenting, learning, and then thinking about creating standards."

With this positioning in mind, I feel like people took this approach with content types - but then we saw fragmentation as a result and we're now working on providing baseline content types as standard content types to stop that fragmentation.

If we encourage people to just build it and think about creating a standard later - how can we avoid this fragmentation issue? Is the fragmentation issue isolated to content types, or a certain type of feature? If it is a "certain type of feature" - can we articulate the criteria for this type of feature as something you should not just build on your own that should begin with an XIP? Is there such a case?

Reviewer 3: @saulmc  :+1: Ok. I think this makes sense to me. 

From XIP5

> xmtp.org is one such entity and the xmtp.org authority ID value is reserved for standard content types only. Standard content types are those that have been adopted through the XIP process.

In this sense only"`standard content types" are submitted as an XRC and other content types are free to operate in a permissionless fashion. Is that how you see it?

Reviewer: Yep, exactly as I see it. Anyone can create, use, or promote a custom content type. The XRC process allows authors to propose those custom content types be adopted as community standards.

Reviewer 5: @jhaaaa it would be interesting to figure out the causes of fragmentation. Why did people develop their own instead of using someone else's that was already available? If the reason is they didn't like the other then that seems like a healthy process of coming up with a consensus. Propose something else and see what wins in the marketplace.

> Parties involved in the process are you, the champion or *XIP author*, the [*XIP editors*](#xip-editors), and the *XMTP Core Developers* (currently the XMTP Labs team).

Reviewer 5: This sounds to me as if the community was excluded from the XIP process.

Author: @mkobetic great feedback. How's this?

```suggestion
Parties involved in the process are you, the champion or *XIP author*, the *XMTP community*, the [*XIP editors*](#xip-editors), and the *XMTP Core Developers* (currently the XMTP Labs team).
```

Reviewer 5: Looks good.

> Once the idea has been vetted (as described in [*Getting Started*](#getting-started)), your next responsibility will be to present (by means of an XIP Draft) the idea to the reviewers and all interested parties, invite editors, developers, and the community to give feedback on the aforementioned channels. You should try and gauge whether the interest in your XIP is commensurate with both the work involved in implementing it and how many parties will have to conform to it. Negative community feedback will be taken into consideration and may prevent your XIP from moving past the Draft stage.

Author: @saulmc Here's a stab:
```suggestion
Vetting your idea within a discussion is a critical part of the XIP process. We encourage sharing the discussion with reviewers, interested parties, editors, developers, and the community to gather feedback.

Additional vetting criteria may include reviewing reference implementations or assessing whether the idea is too specific for broad adoption within the XMTP ecosystem. It’s important to gauge whether the interest in your XIP is commensurate with the effort required to implement it and how many parties will need to conform to it. While the volume of interest by way of discussion is not an explicitly disqualifying characteristic, it may play a role in advancing beyond the idea stage.

After vetting the idea as described above, your next responsibility is to present the idea as a formal XIP Draft by opening a [Pull Request](https://github.com/xmtp/XIPs/pulls).  Following this step, the [XIP Editors](#xip-editors) will review the Draft and make recommendations on its advancement.
```
